react-events-and-state/src/index.jsx

```diff
@@ -1,57 +1,79 @@
-import React from 'react';
+import React, { useState } from 'react';
 import ReactDOM from 'react-dom';
 
-class CustomButton extends React.Component {
-  // FIRST instantiate a CustomButton object from this class
-  constructor(props) {
-    super(props);
-    this.state = { isClicked: false };
+function CustomButton() {
+  const [buttonTxt, setText] = useState(() => {
+    return 'Click Me!';
+  });
+  const [clicked, setClick] = useState(() => {
+    return false;
+  });
 
-    this.handleClick = this.handleClick.bind(this);
-    // bind creates a permanent this object
-    // create a bound copy of the prototype method handleClick
-    // assign the bound copy to an "own" property of this object
-  }
-
-  handleClick() {
-    // decide how to update state in event listeners
-    // please transition to a new state
-    this.setState({ isClicked: true });
-    // schedules a re-render with a new state
-  }
-
-  // NEXT React immediately call the render method to get the the React elements
-  render() {
-    // 100% decide on how the UI should look
-    if (this.state.isClicked === true) {
-      return (
-        <button onClick={this.handleClick}>Thanks!</button>
-        /*
-        {
-          type: 'button,
-          props: {
-            children: 'Thanks!'
-          }
-        }
-        */
-      );
+  function handleClick() {
+    if (clicked) {
+      setText('Click Me!');
+      setClick(false);
     } else {
-      return (
-        <button onClick={this.handleClick}>Click Me!</button>
-        /*
-        {
-          type: 'button,
-          props: {
-            onClick: this.handleClick,
-            children: 'Click Me!'
-          }
-        }
-        */
-      );
+      setText('Thank you!');
+      setClick(true);
     }
   }
+  return (
+    <button onClick={handleClick}>{buttonTxt}</button>
+  );
```

Comment:
You can use a conditional here to render either button. You could also use a ternary expression to pick the text before your return statement

```js
const text = clicked ? 'Thank you!' : 'Click Me!';
```